src/rulesets/LinearAlgebra/uniformscaling.jl

```diff
@@ -0,0 +1,93 @@
+
+#####
+##### constructor
+#####
+
+function rrule(::Type{T}, x::Number) where {T<:UniformScaling}
+    UniformScaling_back(dx) = (NoTangent(), ProjectTo(x)(unthunk(dx).λ))
+    return T(x), UniformScaling_back
+end
+
+#####
+##### `+`
+#####
+
+function frule((_, Δx, ΔJ), ::typeof(+), x::AbstractMatrix, J::UniformScaling)
+    return x + J, Δx + ΔJ
```

Comment:
Should we expect this to work, or is there a better solution? It doesn't at the moment because we can't add `Tangent{P}` to anything that isn't `P`.

a very hacky workaround would be
```suggestion
    return x + J, Δx + (zero(J) + ΔJ)
```
but we might want to define a projection onto a matrix?

Reply:
I think the hack is good actually. It shouldn't be expensive.

I am not too sure why a `Tangent{UniformScaling}` exists at all. I suspect that CRTU doesn't yet know that it shouldn't make one, as I'm sure it knows never to test with say `Tangent{Complex}`. Maybe that's an example of David's point?

Reply:
`rand_tangent` controls this in CRTU. Re David's point, CR is always expected to be loaded when AD is running.